doc/qos/ECN_and_WRED_statistics_HLD.md

```diff
@@ -0,0 +1,342 @@
+
+#### Changes in COUNTERS_DB
+
+The following new port counters will be added along with existing counters on supported platforms
+
+* COUNTERS_PORT_NAME_MAP:port
+    * SAI_PORT_STAT_GREEN_WRED_DROPPED_PACKETS
+    * SAI_PORT_STAT_YELLOW_WRED_DROPPED_PACKETS
+    * SAI_PORT_STAT_RED_WRED_DROPPED_PACKETS
+    * SAI_PORT_STAT_WRED_DROPPED_PACKETS
+
+For every egress queue, the following statistics will be added along with existing queue conters on supported platforms
+
+* COUNTERS_QUEUE_PORT_MAP:queue
+    * SAI_QUEUE_STAT_WRED_ECN_MARKED_PACKETS
+    * SAI_QUEUE_STAT_WRED_ECN_MARKED_BYTES
+
+#### Changes in FLEX_COUNTER_DB
+
+There are no new flex counter groups introduced for this feature.
+
+On supported platforms,
+* The ECN queue counters will use existing flexcounter group QUEUE_STAT_COUNTER_FLEX_COUNTER_GROUP
+* The WRED port counters will use the existing flex counter group PORT_STAT_COUNTER_FLEX_COUNTER_GROUP.
+
+#### Changes in STATE_DB
+State DB will store information about ECN and WRED counter support as per the platform capability. This information will be populated during Orchagent startup by checking the platform capability.
+
+```
+"QUEUE_COUNTER_CAPABILITIES": {
+    "QUEUE_ECN_MARKED_PKT_COUNTER": {
+        "isSupported": "true",
+    },
+    "QUEUE_ECN_MARKED_BYTE_COUNTER": {
+       "isSupported": "true",
+    },
+}
+
+"PORT_COUNTER_CAPABILITIES": {
+    "PORT_WRED_GREEN_DROP_COUNTER": {
+        "isSupported": "true",
+    },
+    "PORT_WRED_YELLOW_DROP_COUNTER": {
+        "isSupported": "true",
+    },
+    "PORT_WRED_RED_DROP_COUNTER": {
+        "isSupported": "true",
+    },
+    "PORT_WRED_TOTAL_DROP_COUNTER": {
+        "isSupported": "true",
+    },
+}
+
+```
+
+The default capability will be isSupported=false for all the above statistics.
+
+### Changes in Orchagent
+Orchagent gets the WRED and ECN counter capability during the startup and updates the STATE_DB with supported statistics. If a counter is supported, respective capability will be set to true. Otherwise the capability will be set to false.  Based on the capability in STATE_DB, FLEXCOUNTER_DB will be updated with supported statistics for polling.
+<p align=center>
+<img src="ecn-wred-stats-images/orchagent_db_state_flow.png" alt="StateDB syncd interactions">
+</p>
+
+### CLI Changes
+
+There are no changes to existing CLI commands. However, the output of existing CLI would change. The CLI will display the ECN-WRED statistics only if the capability is supported by the platform. It gets the capability from STATE_DB and queries only the supported statistics from COUNTERS_DB.
+Following are the CLI usages,
+
+* ECN Marked Statistics
+    * Cleared on user request (clear queue) :  clear queuecounters
+    * Used by CLI (show queue) : show queue counters [interface-name]
+* WRED drop statistics
+    * Cleared on user request (clear counters) :  clear counters
+    * Used by CLI (show queue) : show interfaces counters detailed \<interface-name\>

```

Comment:
used by cli(show interfaces)

Reply:
Yes. Updated the usage of CLIs in new HLD. Thank you!